skyportal/models/classification.py

```diff
@@ -65,3 +71,53 @@ class Classification(Base):
         passive_deletes=True,
         doc="Groups that can access this Classification.",
     )
+    votes = relationship(
+        'ClassificationVote',
+        back_populates="classification",
+        cascade="save-update, merge, refresh-expire, expunge",
+        passive_deletes=True,
+        doc="Classification votes for this classification.",
+    )
+
+
+class ClassificationVote(Base):
+    """Record of an instance in which a Classification is voted up or down
+    (as noted by thumbs up or down from the User on the Source page).
+    """
+
+    create = read = update = delete = AccessibleIfRelatedRowsAreAccessible(
+        classification='read'
+    )
+
+    classification_id = sa.Column(
+        sa.ForeignKey('classifications.id', ondelete='CASCADE'),
+        nullable=False,
+        unique=False,
+        index=True,
+        doc="Object ID for which the vote was registered.",
+    )
+    classification = relationship(
+        'Classification',
+        back_populates='votes',
+        doc='The Classification the vote is associated with.',
+    )
+    voter_id = sa.Column(
+        sa.ForeignKey('users.id', ondelete='CASCADE'),
+        nullable=False,
+        index=True,
+        doc="ID of the User that made this ClassificationVote",
+    )
+    voter = relationship('User', doc="The User that scanned this source.")
+    created_at = sa.Column(
+        sa.DateTime,
+        nullable=False,
+        default=datetime.utcnow,
+        index=True,
+        doc="UTC timestamp of the vote.",
+    )
+    vote = sa.Column(
+        sa.Integer,
+        doc='User-assigned vote for classification',
```

Comment:
I assume this will be +1 or -1? Maybe the doc should say that.

Reply:
yes.